includes/rest/class-replies-controller.php

```diff
@@ -90,29 +100,104 @@ public function get_items( $request ) {
 				\sprintf(
 					// translators: %s: The type (post, comment, etc.) for which no replies collection exists.
 					\__( 'No reply collection exists for the type %s.', 'activitypub' ),
-					$type
+					$object_type
 				),
 				array( 'status' => 404 )
 			);
 		}
 
+		switch ( $request->get_param( 'type' ) ) {
+			case 'replies':
+				$response = $this->get_replies( $request, $wp_object );
+				break;
+
+			case 'likes':
+				$response = $this->get_likes( $request, $wp_object );
+				break;
+
+			case 'shares':
+				$response = $this->get_shares( $request, $wp_object );
+				break;
+
+			default:
+				$response = new \WP_Error( 'rest_unknown_collection_type', 'Unknown collection type.', array( 'status' => 404 ) );
+		}
+
+		// Prepend ActivityPub Context.
+		$response = array_merge( array( '@context' => Base_Object::JSON_LD_CONTEXT ), $response );
+		$response = \rest_ensure_response( $response );
+		$response->header( 'Content-Type', 'application/activity+json; charset=' . \get_option( 'blog_charset' ) );
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves a collection of replies.
+	 *
+	 * @param \WP_REST_Request     $request   The request object.
+	 * @param \WP_Post|\WP_Comment $wp_object The WordPress object.
+	 *
+	 * @return array Response collection of replies.
+	 */
+	public function get_replies( $request, $wp_object ) {
 		$page = $request->get_param( 'page' );
 
 		// If the request parameter page is present get the CollectionPage otherwise the Replies collection.
-		if ( $page ) {
+		if ( $page <= 1 ) {
+			$response = Replies::get_collection( $wp_object );
+		} else {
 			$response = Replies::get_collection_page( $wp_object, $page );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves a collection of likes.
+	 *
+	 * @param \WP_REST_Request     $request   The request object.
+	 * @param \WP_Post|\WP_Comment $wp_object The WordPress object.
+	 *
+	 * @return array Response collection of likes.
+	 */
+	public function get_likes( $request, $wp_object ) {
+		if ( ! $wp_object instanceof \WP_Post ) {
+			$likes = 0;
 		} else {
-			$response = Replies::get_collection( $wp_object );
+			$likes = Interactions::count_by_type( $wp_object->ID, 'like' );
 		}
```

Comment:
In cases like this, where we have something positive to test for, let's do that in the if case and fall back to the default in the else statement.
Alternatively you could define the default and then just do the if case, to change it if the check succeeds.